src/main.cpp

```diff
@@ -203,6 +206,42 @@ class BroadcastMessageDispatcher : public AbstractTask<3 * configMINIMAL_STACK_S
     }
 };
 
+class BroadcastIPTask : public AbstractTask<3 * configMINIMAL_STACK_SIZE> {
+  public:
+    BroadcastIPTask(const char* taskName,
+                    UBaseType_t priority,
+                    IBSP& bsp,
+                    INetworkManager& networkManager,
+                    ILogger& logger) :
+        AbstractTask(taskName, priority),
+        m_bsp(bsp),
+        m_networkManager(networkManager),
+        m_logger(logger) {}
+    ~BroadcastIPTask() override = default;
+
+  private:
+    IBSP& m_bsp;
+    INetworkManager& m_networkManager;
+    ILogger& m_logger;
+    void task() override {
+        while (true) {
+            // Wait for device to be connected and having a valid id
+            if (m_bsp.getHiveMindUUID() == 0 ||
+                m_networkManager.getNetworkStatus() != NetworkStatus::Connected) {
+                Task::delay(500);
+                continue;
+            }
+            Task::delay(5000);
+            auto& broadcastQueue = MessageHandlerContainer::getBroadcastOutputQueue();
+            IPDiscoveryDTO ipDiscoveryDto(m_networkManager.getSelfIP());
+            MessageDTO message(m_bsp.getHiveMindUUID(), 0, ipDiscoveryDto);
+            broadcastQueue.push(message);
```

Comment:
You could create the MessageDTO in the upper scope at the root of the task, so its prevents recreation on everyloop. Though you would need to know the UUID, so that would require 2 while I guess

Reply:
Changed it so that it creates it in the upper scope and only updates the source id if it differs from the one in bsp.